### ISSUE TITLE
@Builder.Default fails for init expressions using super or this

### ISSUE THREAD

**Issue by janrieke** [rzwitserloot/lombok#1749]
Using a `@Builder` on a class with `@Builder.Default` on a instance variable fails if the initializing expression refers to `this` or uses the keyword `super`.

**Example:**
```java
class Superclass {
	int number = 1;
}

@Builder
class Subclass extends Superclass {
	@Builder.Default
	int number = super.number + 1;
}
```
**Expected result:** Compiles, and the initializing expression refers to `number` of `Superclass`.

**Actual result:** Compiler error: `Cannot use super in a static context`

The problem is caused by `@Builder` moving the initializing expression to a `private static int $default$number()` method, which changes the applicability of `this` and `super`.

**Comment by rzwitserloot** [comment]
This is by design and cannot be fixed. To wit:

The way Builder.Default works is that the builder 'gathers' all the required fields, and once its done that, invokes the all-args constructor. During the 'gather' phase, we don't have an instance.

The only way to fix it is to generate a combinatorial explosion of constructors, to generate a constructor with a boolean for EVERY field in addition to the field value, or to have a sentinel 'I have no value, go with the default' option so that the constructor itself chooses whether to invoke the 'produce default value for this field' method; that's the only way we can make those methods non-static.

All possible options to address this cause more harm than this problem, therefore, they're all a bad idea.

In other words, your concern is valid but addressing it breaks more stuff.
